jwst/refpix/tests/test_refpix.py

```diff
@@ -0,0 +1,675 @@
+       in the averaging of the reference pixels to be subtracted.'''
+
+    # create input data
+    # create model of data with 0 value array
+    ngroups = 5
+    ysize = 1024
+    xsize = 1032
+
+    # make ramp model
+    im = make_rampmodel(ngroups, ysize, xsize)
+
+    im.data = im.data * 10
+
+    # set reference pixel values left and right side, odd and even rows.
+    im.data[:, 1:, :, :4] = 1.0
+    im.data[:, 1:, :, 1028:] = 2.0
+    im.data[0, 3, 50, 3] = 35.0
+
+    # set reference pixels to 'REFERENCE_PIXEL'
+    im.pixeldq[:, :4] = dqflags.pixel['REFERENCE_PIXEL']
+    im.pixeldq[:, 1028:] = dqflags.pixel['REFERENCE_PIXEL']
+
+    # run the step
+    out = RefPixStep.call(im)
+
+    # average reference pixel value should be 1.5 (all 1's on left, all 2's on right)
+    assert(out.data[0, 3, 50, 7] == 28.5)
+
+
+def test_nan_refpix():
+    '''Verify that the reference pixels flagged DO_NOT_USE are not used in the calculation'''
+
+    # Test that flagging a reference pixel with DO_NOT_USE does not use the pixel in the
+    # average. Set the pixel to NaN, which results in a NaN average value if used. If the test
+    # passes, then the NaN was correctly flagged and rejected from the average.
+
+    # create input data
+    # create model of data with 0 value array
+    ngroups = 5
+    ysize = 1024
+    xsize = 1032
+
+    # make ramp model
+    im = make_rampmodel(ngroups, ysize, xsize)
+
+    im.data = im.data * 10
+
+    # set reference pixel values left and right side, odd and even rows.
+    im.data[:, 1:, :, :4] = 1.0
+    im.data[:, 1:, :, 1028:] = 2.0
+    im.data[0, 3, 50, 3] = np.nan
+
+    # set reference pixels to 'REFERENCE_PIXEL'
+    im.pixeldq[:, :4] = dqflags.pixel['REFERENCE_PIXEL']
+    im.pixeldq[:, 1028:] = dqflags.pixel['REFERENCE_PIXEL']
+    im.pixeldq[50, 3] = dqflags.pixel['DO_NOT_USE']
+
+    # run the step
+    out = RefPixStep.call(im)
+
+    # average reference pixel value should be 1.5 (all 1's on left, all 2's on right)
+    assert(out.data[0, 3, 50, 7] == 28.5)
+
+
+def test_do_corrections_subarray_no_oddEven(setup_subarray_cube):
+    '''Test all corrections for subarray data with no even/odd.'''
+
+    # Create inputs and subarray SUB320A335R data, and set correction parameters
+    ngroups = 5
+    nrows = 160
+    ncols = 160
+    xstart = 1
+    ystart = 1
+
+
+    odd_even_columns = False
+    use_side_ref_pixels = False
+    side_smoothing_length = 11
+    side_gain = 1.0
+    odd_even_rows = False
+
+    left_rpix = 5
+    bottom_rpix = 7
+    dataval = 150
+    rmean = np.mean([left_rpix, bottom_rpix])
+
+    input_model = setup_subarray_cube('SUB160', 'NRCB1', xstart, ystart, ngroups, nrows, ncols)
+    input_model.data[0, 0, :, :] = dataval
+    input_model.data[0, 0, :4, :] = bottom_rpix
+    input_model.data[0, 0, :, :4] = left_rpix
+    input_model.pixeldq[:4, :] = 2147483648
+    input_model.pixeldq[:, :4] = 2147483648
```

Comment:
@mfixstsci @jdavies-st  Sorry, I should have written `dqflags.pixel['REFERENCE_PIXEL'])` instead of the `2147483648` value. I pulled that out of some older code.